src/OrchardCore/OrchardCore.Users.Core/Services/UserStore.cs

```diff
@@ -396,15 +455,17 @@ public async Task AddToRoleAsync(IUser user, string normalizedRoleName, Cancella
                 throw new ArgumentNullException(nameof(user));
             }
 
-            var roleNames = await _roleService.GetRoleNamesAsync();
```

Comment:
The same way you created a RoleStore that does nothing, you can create a default RoleService that does what is necessary when the Roles module is not enabled.